File: Arrays_&_Hashing/Top_K_Frequent_Elements/2_Better.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> freq;
        for(int n: nums) freq[n]++;
        vector<pair<int,int>> vec;
        for(auto &p: freq) vec.push_back({p.second,p.first});
        sort(vec.begin(), vec.end(), greater<pair<int,int>>());
        vector<int> result;
        for(int i=0;i<k;i++) result.push_back(vec[i].second);
        return result;
    }
};

File: Arrays_&_Hashing/Top_K_Frequent_Elements/2_Better.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        java.util.Map<Integer,Integer> freq = new java.util.HashMap<>();
        for(int n: nums) freq.put(n, freq.getOrDefault(n,0)+1);
        java.util.List<java.util.Map.Entry<Integer,Integer>> list = new java.util.ArrayList<>(freq.entrySet());
        list.sort((a,b)->b.getValue()-a.getValue());
        int[] ans = new int[k];
        for(int i=0;i<k;i++) ans[i]=list.get(i).getKey();
        return ans;
    }
}
